codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoder.java

```diff
@@ -153,7 +152,15 @@ public ChannelFuture writeHeaders(final ChannelHandlerContext ctx, final int str
         try {
             Http2Stream stream = connection.stream(streamId);
             if (stream == null) {
-                stream = connection.local().createStream(streamId, endOfStream);
+                try {
+                    stream = connection.local().createStream(streamId, endOfStream);
+                } catch (Throwable cause) {
```

Comment:
Currently it looks like we should be able todo this ... however if the implementation changes and any other exception type escapes (e.g. `IllegalState`) we will send a GO_AWAY. Maybe this is OK though?